File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int sm = 0;
        int ans = INT_MIN;
        int mx = INT_MIN;
        for (int i=0; i<n; i++) {
            sm += nums[i];
            if (sm < 0) {
                sm = 0;
            }
            mx = max(mx, nums[i]);
            ans = max(ans, sm);
        }
        if (mx < 0) {
            return mx;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/permutations/submission-0.java
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    public List<List<Integer>> permute(int[] nums) {
        List<Integer> v = new ArrayList<>();
        boolean[] pick = new boolean[nums.length];
        dfs(nums, v, pick);
        return ans;
    }

    void dfs(int[] nums, List<Integer> v, boolean[] pick) {
        if (v.size() == nums.length) {
            ans.add(new ArrayList<>(v));
            return;
        }

        // take
        for (int j=0; j<nums.length; j++) {
            if (!pick[j]) {
                v.add(nums[j]);
                pick[j] = true;
                dfs(nums, v, pick);
                v.removeLast();
                pick[j] = false;
            }
        }
    }
}
